app/controllers/draws_controller.rb

```diff
@@ -165,36 +165,39 @@ def set_draw
   end
 
   def calculate_metrics
+    calculate_sizes
     calculate_suite_metrics
     calculate_group_metrics
     calculate_oversub_metrics
     calculate_ungrouped_students_metrics
   end
 
+  def calculate_sizes
+    @suite_sizes ||= @draw.suite_sizes
+    @group_sizes ||= @draw.group_sizes
+    @sizes ||= (@suite_sizes + @group_sizes).uniq.sort
+  end
+
   def calculate_suite_metrics
-    @suite_sizes = SuiteSizesQuery.new(@draw.suites.available).call
     @suite_counts = @draw.suites.available.group(:size).count
     @suite_counts.default = 0
   end
 
   def calculate_group_metrics
-    @draw_sizes = DrawSizesQuery.call(draw: @draw)
-    empty_groups_hash = @draw_sizes.map { |s| [s, []] }.to_h
     @groups = @draw.groups.includes(:leader)
-    @groups_by_size =
-      empty_groups_hash.merge(@groups.sort_by { |g| Group.statuses[g.status] }
-                                     .group_by(&:size))
-    @groups_by_size.default = 0
-    @group_sizes = @groups_by_size.keys
+    @groups_by_size = @groups.sort_by { |g| Group.statuses[g.status] }
+                             .group_by(&:size)
+    @groups_by_size.default = []
   end
 
-  def calculate_oversub_metrics # rubocop:disable AbcSize
+  def calculate_oversub_metrics
     return unless policy(@draw).oversub_report?
-    zeroed_group_hash = @group_sizes.map { |s| [s, 0] }.to_h
-    @group_counts = zeroed_group_hash.merge(@groups.group(:size).count)
-    @locked_counts = zeroed_group_hash.merge(@groups.where(status: 'locked')
-                                                    .group(:size).count)
-    @diff = @group_sizes.map do |size|
+    calculate_sizes
+    @group_counts = @groups.group(:size).count
+    @group_counts.default = 0
+    @locked_counts = @groups.where(status: 'locked').group(:size).count
+    @locked_counts.default = 0
+    @diff = @sizes.map do |size|
```

Comment:
Minor,  but do we want to replace this with `each_with_object` to avoid `to_h`? Now that we've found it I think it reads much better.